src/server/main_service.cc

```diff
@@ -553,7 +553,7 @@ optional<Transaction::MultiMode> DeduceExecMode(ExecEvalState state,
     // Allow multi eval only when scripts run global and multi runs in global or lock ahead
     // We adjust the atomicity level of multi transaction inside StartMultiExec i.e if multi mode is
     // lock ahead and we run global script in the transaction then multi mode will be global.
-    if (!contains_global || (multi_mode == Transaction::NON_ATOMIC)) {
+    if ((multi_mode == Transaction::NON_ATOMIC)) {
       return nullopt;
     }
```

Comment:
1. Comment is now outdated
2. I think it'll work for non atomic too, why the exception 🤔